### PR TITLE
Disk storage prototype

### PR DESCRIPTION
**Pull request by eadventurous**
### Description of the Change

Resolves issue: [RH2-1] Implement Disk Storage prototype 

1. File structure improvements
2. Reading from file
3. Async disk read and write with **async_std**
4. Disk storage tests now cleanup folders after themselves

### Benefits

Introduces the prototype of disk storage. Right now as agreed 1 block = 1 file.

### Possible Drawbacks 

Dependency on async_std.

Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### CODE REVIEW

File: src/storage/kura.rs [hunk 1]
```diff
@@ -170,43 +173,122 @@ impl MerkleTree {
     }
 }
 
+static DEFAULT_BLOCK_STORE_LOCATION: &str = "./blocks/";
```
Comment (humb1t) [inline src/storage/kura.rs]: It's better to hardcode this parameter for now and remove `'a`.

Looks quite dirty right now - we have a lot of unnecessary lifetime specifiers to use a string in one place.

File: src/storage/kura.rs [hunk 2]
```diff
@@ -170,43 +173,122 @@ impl MerkleTree {
...
+        Disk::new(DEFAULT_BLOCK_STORE_LOCATION)
+    }
+}
+
+impl<'a> Disk<'a> {
+    fn new(block_store_location: &'a str) -> Disk {
+        use std::fs;
+
+        let path = Path::new(block_store_location);
+        fs::create_dir_all(path).expect("Failed to create storage directory");
```
Comment (humb1t) [inline src/storage/kura.rs]: add `.` in the end of a sentence.

File: src/storage/kura.rs (hunk 1 above)
Comment (eadventurous) [reply src/storage/kura.rs]: If I remove `block_store_location` field, the logic with cleaning temporary up folders will also have to be removed. Not sure if it's a good idea.

Comment (eadventurous) [reply src/storage/kura.rs]: Maybe we can make it into an owned `String` and owned `PathBuf` to remove the lifetime specifiers. What do you think @humb1t ?

Comment (humb1t) [reply src/storage/kura.rs]: @eadventurous let's start from owned solution

**Pull request merged**